src/analysis/souffle/auth_logic_interface.dl

```diff
@@ -0,0 +1,34 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+// This file defines the interface between authorization logic and taint 
+// tracking / information flow control.
+
+#ifndef AUTH_LOGIC_INTERFACE 
+#define AUTH_LOGIC_INTERFACE
+
+// A Principal that makes assertions in authorization logic
+.type Principal <: symbol
+
+// Some party asserts the claim that an accessPath should be downgraded from
+// one Tag to another.
+.decl saysDowngrades(speaker: Principal, p: AccessPath, t: Tag)
```

Comment:
This relation is called `saysDowngrades`, which implies a sort of power hierarchy on the defined tags. However, I don't see any sort of hierarchy defined upon the tags in these files. Should this instead be something like `saysRetag`? Or is the hierarchy coming in a later commit?

Reply:
@bgogul pointed me to what this means, ignore this comment.